projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts

```diff
@@ -228,173 +210,81 @@ describe('IgxHierarchicalGrid Integration #hGrid', () => {
-    //         fixture.detectChanges();
-
-    //         const childGrids =  fixture.debugElement.queryAll(By.css('igx-child-grid-row'));
-    //         const childGrid = childGrids[0].query(By.css('igx-hierarchical-grid')).componentInstance;
-    //         expect(childGrid.data).toBe(fixture.componentInstance.data[1]['childData']);
-
-    //     }));
-    // });
-
     describe('Filtering', () => {
-        it('should enable filter-row for root and child grids', fakeAsync(/** filter showHideArrowButtons + row toggle rAF */() => {
-            let filteringCells = fixture.debugElement.queryAll(By.css('igx-grid-filtering-cell'));
 
+        it('should enable filter-row for root and child grids', fakeAsync(() => {
+            let filteringCells = fixture.debugElement.queryAll(By.css(FILTERING_CELL_CLASS));
             expect(filteringCells.length).toEqual(3);
-            filteringCells[0].query(By.css('igx-chip')).nativeElement.click();
-            fixture.detectChanges();
-            expect(document.querySelectorAll('igx-grid-filtering-row').length).toEqual(1);
 
-            const firstRow = hierarchicalGrid.dataRowList.toArray()[0];
-            // first child of the row should expand indicator
-            (firstRow.nativeElement.children[0] as HTMLElement).click();
-            fixture.detectChanges();
+            GridFunctions.clickFilterCellChipUI(fixture, 'ID');
+            expect(document.querySelectorAll(FILTERING_ROW_CLASS).length).toEqual(1);
+
+            // expand first row
+            hierarchicalGrid.expandRow(hierarchicalGrid.dataRowList.first.rowID);
 
-            filteringCells = fixture.debugElement.queryAll(By.css('igx-grid-filtering-cell'));
+            filteringCells = fixture.debugElement.queryAll(By.css(FILTERING_CELL_CLASS));
             expect(filteringCells.length).toEqual(6);
-            filteringCells[3].query(By.css('igx-chip')).nativeElement.click();
-            fixture.detectChanges();
-            expect(document.querySelectorAll('igx-grid-filtering-row').length).toEqual(2);
+
+            GridFunctions.clickFilterCellChipUI(fixture, 'ProductName', hierarchicalGrid.hgridAPI.getChildGrids(false)[0]);
+            expect(document.querySelectorAll(FILTERING_ROW_CLASS).length).toEqual(2);
         }));
 
         it('should not lose child grid states after filtering in parent grid.', fakeAsync(() => {
-            // expand 1st row
-            (hierarchicalGrid.dataRowList.toArray()[0].nativeElement.children[0] as HTMLElement).click();
-            fixture.detectChanges();
-            const childGrids =  fixture.debugElement.queryAll(By.css('igx-child-grid-row'));
+            // expand first row
+            hierarchicalGrid.expandRow(hierarchicalGrid.dataRowList.first.rowID);
+            const childGrids = fixture.debugElement.queryAll(By.css('igx-child-grid-row'));
             let childGrid = childGrids[0].query(By.css('igx-hierarchical-grid')).componentInstance;
```

Comment:
You can create a test method to get the child grids as it it used in many tests

Reply:
Nah it's always to get the first child which is perfectly achievable with ```hierarchicalGrid.hgridAPI.getChildGrids(false)[0];```

I can shorthand it through a function but it doesn't seem that necessary. The hgrid API should be transparent enough.

Reply:
This was just left like this because I am apparently blind :)